### PR TITLE
adding password recovery with multiple challenge questions at once

### PR DESCRIPTION
**Pull request by GayanM**

### CODE REVIEW

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/serviceclient/beans/ErrorResponse.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (harsha1979) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/serviceclient/beans/ErrorResponse.java]: Old license header

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/challenge-question-process.jsp [hunk 2]
```diff
@@ -41,9 +44,20 @@
         User user = IdentityManagementServiceUtil.getInstance().getUser(userName);
         session.setAttribute("user", user);
         PasswordRecoverySecurityQuestionClient pwRecoverySecurityQuestionClient = new PasswordRecoverySecurityQuestionClient();
-        ChallengeQuestionResponse challengeQuestionResponse = pwRecoverySecurityQuestionClient.initiateUserChallengeQuestion(user);
-        session.setAttribute("challengeQuestionResponse", challengeQuestionResponse);
-        request.getRequestDispatcher("challenge-question-view.jsp").forward(request, response);
+        Response response1 = pwRecoverySecurityQuestionClient.initiateUserChallengeQuestion(user);
```
Comment (harsha1979) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/challenge-question-process.jsp]: response1 ?

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/challenge-question-request.jsp [hunk 3]
```diff
@@ -26,68 +26,47 @@
...
         if (Boolean.parseBoolean(application.getInitParameter(
                 IdentityManagementEndpointConstants.ConfigConstants.PROCESS_ALL_SECURITY_QUESTIONS))) {
-            UserChallengesCollectionDTO userChallengesCollectionDTO =
-                    userInformationRecoveryClient.getChallengeQuestions(username, confirmationKey);
-            userChallengesDTOs = userChallengesCollectionDTO.getUserChallengesDTOs();
-
-            if (ArrayUtils.isEmpty(userChallengesDTOs)) {
+            User user = IdentityManagementServiceUtil.getInstance().getUser(username);
+            session.setAttribute("user", user);
+            Response response1 = pwRecoverySecurityQuestionClient.initiateUserChallengeQuestionAtOnce(user);
```
Comment (harsha1979) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/challenge-question-request.jsp]: response1 ?

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/challenge-question-request.jsp [hunk 4]
```diff
@@ -26,68 +26,47 @@
...
+
+            int statusCode = response1.getStatus();
+            if(Response.Status.OK.getStatusCode() == statusCode) {
+                ChallengeQuestionsResponse challengeQuestionsResponse = response1.readEntity(ChallengeQuestionsResponse.class);
+                session.setAttribute("challengeQuestionsResponse", challengeQuestionsResponse);
+                challengeQuestions = challengeQuestionsResponse.getQuestion();
+            } else if (Response.Status.BAD_REQUEST.getStatusCode() == statusCode || Response.Status.INTERNAL_SERVER_ERROR.getStatusCode() == statusCode) {
+                ErrorResponse errorResponse = response1.readEntity(ErrorResponse.class);
                 request.setAttribute("error", true);
-                request.setAttribute("errorMsg",
```
Comment (harsha1979) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/challenge-question-request.jsp]: errorMsg, error can be constant

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/challenge-question-request.jsp [hunk 5]
```diff
@@ -26,68 +26,47 @@
...
-                request.getRequestDispatcher("processsecurityquestions.do").forward(request, response);
-                return;
-            } else {
-                request.setAttribute("error", true);
-                request.setAttribute("errorMsg",
-                                     "Could not find Security Questions. Seems you have not configured them.");
-                request.getRequestDispatcher("error.jsp").forward(request, response);
-                return;
-            }
+            request.getRequestDispatcher("challenge-question-process.jsp?username=" + username).forward(request,
```
Comment (harsha1979) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/challenge-question-request.jsp]: username should be constant

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/serviceclient/beans/ErrorResponse.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+package org.wso2.carbon.identity.mgt.endpoint.serviceclient.beans;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlRootElement;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "")
+public class ErrorResponse {
```
Comment (omindu) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/serviceclient/beans/ErrorResponse.java]: Missing class level comment.

**Pull request merged**